src/rust/ensogl/lib/core/src/control/io/mouse/event.rs

```diff
@@ -47,6 +49,37 @@ macro_rules! define_events {
             pub fn button(&self) -> mouse::Button {
                 mouse::Button::from_code(self.raw.button().into())
             }
+
+            /// Return the position relative to the event handler that was used to catch the event.
+            /// If the event handler does not have a position in the DOM, the returned position
+            /// will be relative to the viewport.
+            ///
+            /// Note: may cause reflow of the JS layout.
+            pub fn position_in_event_handler(&self) -> Vector2<f32> {
+                if let Some(element) = self.get_event_source_element() {
+                      self.relative_position(&element)
+                } else {
+                    Vector2::new(self.client_x() as f32,self.client_y() as f32)
+                }
+            }
+
+            /// Return the event handler that caught this event, if it exists and if it is a
+            /// html element. Returns `None`, if the event was caught e.g., byt the window.
+            fn get_event_source_element(&self) -> Option<web_sys::Element> {
```

Comment:
I'm not really ok with defining so many new names for things that actually have a name already. This, for example, should be named `get_current_target` instead.